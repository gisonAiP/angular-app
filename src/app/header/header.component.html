<header data-test="header">
  <div class="header-desktop d-none d-md-flex">
    <div class="header-logo">
      <img src="./assets/svg/tripico.svg" alt="logo" height="33"/>
    </div>
    <input class="header-search" placeholder="Where are you going?"/>
    <div class="header-icons-container">
      <img src="./assets/svg/home.svg" alt="home" height="20"/>
      <img src="./assets/svg/briefcase.svg" alt="briefcase" height="20"/>
      <img src="./assets/svg/compass.svg" alt="compass" height="20"/>
      <img src="./assets/svg/repeat.svg" alt="repeat" height="20"/>
      <img src="./assets/svg/avatar.svg" alt="avatar" height="20"/>
    </div>
  </div>
  <div class="header-mobile d-flex d-md-none justify-content-between py-3 px-2">
    <div>
      <p class="title">Marketplace</p>
    </div>
    <div>
      <img src="./assets/svg/search2.svg" alt="search" height="30" class="px-1"/>
      <img src="./assets/svg/user.svg" alt="user" height="30" class="px-1"/>
      <img src="./assets/svg/bookmark.svg" alt="bookmark" height="30" class="px-1"/>
    </div>
  </div>
</header>
