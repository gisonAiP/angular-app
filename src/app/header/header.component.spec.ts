import { createRoutingFactory, Spectator } from '@ngneat/spectator/jest';
import { HeaderComponent } from './header.component';

describe('HeaderComponent', () => {
  let spectator: Spectator<HeaderComponent>;
  const createComponent = createRoutingFactory({
    component: HeaderComponent,
    shallow: true,
    providers: [],
    declarations: []
  });

  beforeEach(() => {
    spectator = createComponent();
    spectator.detectChanges();
  });

  it('should validate snapshot', () => {
    expect(spectator.fixture).toMatchSnapshot();
  });
});

