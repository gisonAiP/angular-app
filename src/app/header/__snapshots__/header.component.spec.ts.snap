// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent should validate snapshot 1`] = `
<app-header>
  <header
    data-test="header"
  >
    <div
      class="header-desktop d-none d-md-flex"
    >
      <div
        class="header-logo"
      >
        <img
          alt="logo"
          height="33"
          src="./assets/svg/tripico.svg"
        />
      </div>
      <input
        class="header-search"
        placeholder="Where are you going?"
      />
      <div
        class="header-icons-container"
      >
        <img
          alt="home"
          height="20"
          src="./assets/svg/home.svg"
        />
        <img
          alt="briefcase"
          height="20"
          src="./assets/svg/briefcase.svg"
        />
        <img
          alt="compass"
          height="20"
          src="./assets/svg/compass.svg"
        />
        <img
          alt="repeat"
          height="20"
          src="./assets/svg/repeat.svg"
        />
        <img
          alt="avatar"
          height="20"
          src="./assets/svg/avatar.svg"
        />
      </div>
    </div>
    <div
      class="header-mobile d-flex d-md-none justify-content-between py-3 px-2"
    >
      <div>
        <p
          class="title"
        >
          Marketplace
        </p>
      </div>
      <div>
        <img
          alt="search"
          class="px-1"
          height="30"
          src="./assets/svg/search2.svg"
        />
        <img
          alt="user"
          class="px-1"
          height="30"
          src="./assets/svg/user.svg"
        />
        <img
          alt="bookmark"
          class="px-1"
          height="30"
          src="./assets/svg/bookmark.svg"
        />
      </div>
    </div>
  </header>
</app-header>
`;
